Libraries/libutil/Sources/DefaultFilesystem.cpp

```diff
@@ -730,23 +733,53 @@ readSymbolicLink(std::string const &path, bool *directory) const
     WideString target = WideString(&pathBuffer[offset], &pathBuffer[offset + length]);
     return WideStringToString(target);
 #else
-    for (size_t size = PATH_MAX; true; size *= 2) {
```

Comment:
I suggest put this code in its own function and having the iterative logic use that function.

In other words, keep `readSymbolicLink` as-is and introduce `resolveSymbolicLinks` which calls `readSymbolicLink`.

Right now, your logic isn't triggered on Windows (should it?), and I think that's an oversight. Also, this function is pretty complex (doubly nested loops with early returns) after your changes.

Reply:
(As written, reviewing the code thoroughly will make my head hurt.)